tests/behat/features/capabilities/event/eventenrollment.feature

```diff
@@ -87,59 +86,58 @@ Feature: Enroll for an event
     Then I should see "No one has enrolled for this event"
     And I should see the button "Enroll"
     And I should see the link "Manage enrollments"
-
-    When I press the "Enroll" button
+    And I press the "Enroll" button
     And I wait for AJAX to finish
-    Then I should see the text "Meetup: My Behat Event created" in the "Modal"
+    And I should see the text "Meetup: My Behat Event created" in the "Modal"
     And I should see the link "See who else is going" in the "Modal"
     And I press the "Close" button
     And I should see the button "Enrolled"
-    Then I reload the page
+    And I reload the page
     And I should see "1 person has enrolled"
     And I should see the link "All enrollments"
-
-    When I press the "Enrolled" button
+    And I press the "Enrolled" button
     And I press "Cancel enrollment"
     And I wait for AJAX to finish
     And I reload the page
-    Then I should see "No one has enrolled for this event"
+    And I should see "No one has enrolled for this event"
     And I should see the button "Enroll"
     And I should see the link "Manage enrollments"
-
     # Enroll again, since this is technically something different.
-    When I press the "Enroll" button
+    And I press the "Enroll" button
     And I wait for AJAX to finish
     And I press the "Close" button
     And I reload the page
-    Then I should see "1 person has enrolled"
+    And I should see "1 person has enrolled"
     And I should see the link "All enrollments"
 
   @verified @cache
   Scenario: Successfully changed enrollment and see changes in teaser
     Given users:
       | name            | pass            | mail                        | status | roles    |
       | eventenrollment | eventenrollment | eventenrollment@example.com | 1      | verified |
+
     When I am logged in as "eventenrollment"
     And I am viewing my event:
       | title            | Enrollment test event |
       | field_event_date | 3014-10-17 8:00am     |
       | status           | 1                     |
     And I click "eventenrollment" in the "Main content"
     And I click "Events"
-    Then I should not see "Enrolled"
 
-    When I click "Enrollment test event"
+    Then I should not see "Enrolled"
+    And I click "Enrollment test event"
     And I press the "Enroll" button
     And I wait for AJAX to finish
     And I press the "Close" button
-    Then I should see the button "Enrolled"
+    And I should see the button "Enrolled"
     And I click "eventenrollment" in the "Main content"
     And I click "Events"
-    Then I should see "Enrolled"
+    And I should see "Enrolled"
 
   @closed_enrollments
   Scenario: Can no longer enroll to an event when it has finished.
     Given I am logged in as an "verified"
+
     When I am viewing my event:
```

Comment:
@ribel @Macsym2018 I think some more work may be needed in this PR (I realise we just merged it).

While doing some work on our test infrastructure this weekend I found that "When I am viewing my event" doesn't actually validate the event that we create. So one of the things that was missed in this PR is that we actually have test coverage here to make sure that events work without end date. However, this test step is now creating invalid events.

Reply:
Hi @Kingdutch,

We have tested this change on a new installation (Tugboat with Demo content) and also update hook on site with existing events (with and without end dates). It works as expected and matches the acceptance criteria. More details can be found in JIRA ticket.

> When an existing event is edited and saved a validation message should be displayed to require the end date value if not available. 

![image](https://github.com/goalgorilla/open_social/assets/2241917/58b1e10d-9d95-432e-90b4-314d124da6ce)
![image](https://github.com/goalgorilla/open_social/assets/2241917/2b7d074c-f16e-4f38-9ccb-d2107be67007)

Reply:
Hey @ribel , I'm not saying your change is malfunctioning. :) I am saying our test is broken.

a) We're now testing something we don't support (a date without end date) which means our test coverage doesn't reflect how we want our product to work. I'll raise a follow-up PR for this.
b) It shows our test tooling doesn't warn us if we do things that shouldn't work: we shouldn't be able to create an event in an invalid state. I'm working on this in a separate track.